homeassistant/components/openweathermap/weather.py

```diff
@@ -210,9 +210,9 @@ def calc_precipitation(rain, snow):
                             "temp"
                         ),
                         ATTR_FORECAST_PRECIPITATION: (
-                            round(entry.get_rain().get("3h"), 1)
-                            if entry.get_rain().get("3h") is not None
-                            and (round(entry.get_rain().get("3h"), 1) > 0)
+                            round(entry.get_rain().get("1h"), 1)
+                            if entry.get_rain().get("1h") is not None
+                            and (round(entry.get_rain().get("1h"), 1) > 0)
                             else None
```

Comment:
Please write out this without a ternary operator expression. We avoid those when they span more than one line.

Reply:
Sure; I am wondering though, if this is necessary. OWM does not provide the rain field, if it is not raining/there is no rain in the forecast. Thus, I think we can simplify this to
```py
round(entry.get_rain().get("1h"), 1)
```

Reply:
I just changed the numbers without wanting to change any logic code.

Reply:
I'm not following. What are the possible results that `entry.get_rain` returns?

Reply:
1. `None` if no rain is reported in current/forecast
2. A number > 0 for daily forecast. This is already handled [above](https://github.com/home-assistant/core/blob/4e2c49ba2a250aaccfc6ce32398ec2dbebfaef1a/homeassistant/components/openweathermap/weather.py#L185).
3. A dict containing only the key '1h' for current or the hourly forecast. This value for 1h is a number as in case 2.

If there is no rain, OWM will not provide a rain field (case 1). Thus, we will get the value None.
If there is rain, it will be a dict, and the value will be a number greater 0.

Reply:
So if we might get a `None` value we have to guard against that so we don't try to round a `None` value.